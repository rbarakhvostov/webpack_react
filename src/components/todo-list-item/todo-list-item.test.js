import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../app';
import TodoListItem from './todo-list-item';

describe('TodoListItem />', () => {
  const wrapper = mount(<App />);

  it('TodoListItem renders without crashing', () => {
    // eslint-disable-next-line no-shadow
    const wrapper = mount(<TodoListItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Trash button works correctly', () => {
    const event1 = { target: { value: 'Create item 1' } };
    const event2 = { target: { value: 'Create item 2' } };
    wrapper.find('.item-add-form-input').simulate('change', event1);
    wrapper.find('.item-add-form').simulate('submit');
    wrapper.find('.item-add-form-input').simulate('change', event2);
    wrapper.find('.item-add-form').simulate('submit');
    wrapper
      .find('#trash-button')
      .first()
      .simulate('click');
    expect(
      wrapper
        .find('.todo-list-item-label')
        .at(0)
        .text(),
    ).toEqual('Create item 2');
  });

  it('Exclamation button works correctly', () => {
    wrapper.find('#exclamation-button').simulate('click');
    expect(wrapper.find('.todo-list-item').hasClass('important')).toBe(true);
    wrapper.find('#exclamation-button').simulate('click');
    expect(wrapper.find('.todo-list-item').hasClass('important')).toBe(false);
  });

  it('Toggle completion works correctly', () => {
    wrapper.find('.todo-list-item').simulate('click');
    expect(wrapper.find('.todo-list-item-label').hasClass('done')).toBe(true);
    wrapper.find('.todo-list-item').simulate('click');
    expect(wrapper.find('.todo-list-item-label').hasClass('done')).toBe(false);
  });
});
