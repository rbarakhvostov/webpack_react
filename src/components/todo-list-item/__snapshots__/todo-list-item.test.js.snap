// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem /> TodoListItem renders without crashing 1`] = `
<TodoListItem>
  <li
    className="todo-list-item list-group-item"
  >
    <span
      className="todo-list-item-label"
    />
    <div
      className="wrap-todo-list-item-button"
    >
      <button
        className="todo-list-item-button btn btn-outline-secondary btn-sm"
        id="exclamation-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="fa fa-exclamation"
        />
      </button>
      <button
        className="todo-list-item-button btn btn-outline-danger btn-sm"
        id="trash-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="fa fa-trash-o"
        />
      </button>
    </div>
  </li>
</TodoListItem>
`;
